src/main/java/uk/gov/companieshouse/company/metrics/controller/CompanyMetricsController.java

```diff
@@ -34,4 +42,45 @@ public CompanyMetricsController(CompanyMetricsService companyMetricsService) {
                 .orElse(ResponseEntity.notFound().build());
     }
 
+    /**
+     * Post request for company metrics.
+     *
+     * @param  companyNumber  the company number for metrics recalculation
+     * @param  requestBody  the request body containing Instructions to recalculate
+     * @return  no response
+     */
+    @PostMapping ("/company/{company_number}/metrics/recalculate")
+    public ResponseEntity<Void> recalculate(
+             @PathVariable("company_number") String companyNumber,
+             @RequestBody MetricsRecalculateApi requestBody
+    ) throws JsonProcessingException {
+
+        // Check to see if mortgages flag is true then process further
+        if (requestBody != null && requestBody.getMortgage()) {
+            // query the mongodb to get a charges counts from company_mortgages
+            int totalCount  = companyMetricsService.queryCompanyMortgages(companyNumber, "none");
+            int satisfiedCount =   companyMetricsService.queryCompanyMortgages(
+                     companyNumber, "satisfied");
+            int partSatisfiedCount = companyMetricsService.queryCompanyMortgages(
+                     companyNumber, "part-satisfied");
+            String updatedBy =  requestBody.getInternalData() != null
+                      ? requestBody.getInternalData().getUpdatedBy() : null;
+
+            Optional<CompanyMetricsDocument> companyMetricsDocument =
+                    companyMetricsService.get(companyNumber);
+            if (companyMetricsDocument.isPresent()) {
```

Comment:
Sorry, opinionated comment, better to avoid if/else where possible, use .map .orElse etc. I leave it to you to decide :)

Reply:
this is resolved